compiler/circle-opselector/driver/Driver.cpp

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// #include "OpSelector.h"
+
+#include <foder/FileLoader.h>
+
+#include <luci/Importer.h>
+#include <luci/CircleExporter.h>
+#include <luci/CircleFileExpContract.h>
+#include <luci/Import/CircleReader.h>
+
+#include <luci/Profile/CircleNodeID.h>
+
+#include <arser/arser.h>
+#include <vconone/vconone.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+void print_version(void)
+{
+  std::cout << "circle-opselector version " << vconone::get_string() << std::endl;
+  std::cout << vconone::get_copyright() << std::endl;
+}
+
+bool check_input(const std::string &str)
+{
+  bool has_hyphen = false;
+
+  if (str.empty())
+    return false;
+  if (str.at(0) == '-' || str[str.size() - 1] == '-')
+  {
+    std::cerr << "ERROR: Invalid input. Please make sure - is between the numbers" << std::endl;
+    return false;
+  }
+
+  for (char c : str)
+  {
+    if (isdigit(c))
+      continue;
+    else if (has_hyphen && c == '-') // when user enter '-' more than 2.
+    {
+      std::cerr << "ERROR: Too many '-' in str." << std::endl;
+      return false;
+    }
+    else if (c == '-')
+      has_hyphen = true;
+    else // when user enter not allowed character, print alert msg.
+    {
+      std::cerr << "ERROR: To select operator by id, please use these args: [0-9], '-', ','"
+                << std::endl;
+      return false;
+    }
+  }
+  return true;
+}
+
+void split_id_input(const std::string &str, std::vector<int> &by_id)
+{
+  std::istringstream ss;
+  ss.str(str);
+  std::string str_buf;
+
+  while (getline(ss, str_buf, ','))
+  {
+    if (check_input(str_buf)) // input validation
+    {
+      try
+      {
+        if (str_buf.find('-') == std::string::npos) // if token has no '-'
+          by_id.push_back(stoi(str_buf));
+        else // tokenize again by '-'
+        {
+          std::istringstream ss2(str_buf);
+          std::string token;
+          int from_to[2], top = 0;
+
+          while (getline(ss2, token, '-'))
+            from_to[top++] = stoi(token);
```

Comment:
when happends user inputs `-------` ?

Reply:
When user enter `{start number}-{end number}`, I wanted to input all integer between  start to end using by for loop.

So I made array `from_to` to put `start` data in `from_to[0]`, `end` data in `from_to[1]`.

Like `1-5-7`, top can be greater than 2, but multiple '-' inputs (or invalid '-' position like `-1`, `1-`) are handled by `check_input` so I thought it would be fine.

But if it is too ambiguous, I'll use `dsv_to_vector`.

Reply:
> so I thought it would be fine.

Then, it would be better to put assertion codes(using `assert()`) in the top of this function to check conditions so that readers will know, and adding commends will help understanding the codes.

> But if it is too ambiguous

Not 'too', just comments about what this clode block is doing will help

Reply:
If adding assert() is a load at this moment, please add `// TODO do assert check` will do.